app/views/commontator/comments/create.js.erb

```diff
@@ -18,24 +18,28 @@
 %>
 
 <% if @commontator_new_comment.nil? %>
-  $("#<%= id %>").hide();
+$("#<%= id %>").hide();
 
-  $("#<%= id %>-link").fadeIn();
+$("#<%= id %>-link").fadeIn();
 <% else %>
-  $("#<%= id %>").html("<%= escape_javascript(
+$("#<%= id %>").html("<%= escape_javascript(
     render partial: 'form', locals: {
       comment: @commontator_new_comment, thread: @commontator_thread
     }
   ) %>");
 <% end %>
 
 <% if @update_icon %>
-$('#commentsIcon').addClass('new-comment');
+$("#commentsIcon").addClass("new-comment");
+<% end %>
+
+<% if @no_unread_comments %>
+$("#commentsIcon").removeClass("new-comment");
 <% end %>
```

Comment:
According to the flow inside the `update_unread_status` method of the `comments_controller`, we will never encounter the case where both variables (`@update_icon` and `@no_unread_comments` are true), so we won't have the case that we add a CSS class just to then remove it immediately. This would result in a short visual flickering which is unwanted. So, this is not a problem.

But still, maybe we can simplify the logic in the controller such that the frontend only has to deal with one variable determining whether to add the CSS class or do nothing? Or do we really have to *remove* the CSS class because it is set in the `_navbar.html.erb` and we need to get rid of it? If so, why not make the changes such that the navbar doesn't add the class in the first place?

Reply:
The navbar is loaded way before. In the situation described in #573, the teacher finds a yellow new_comments icon in the navbar (correctly so), posts their reply, and then the icon has to turn white. During that process, the navbar does not get 
 reloaded. The alternative to what is done in the PR would be to reload the navbar. But the view we have here is a view of the commontator gem which we modify (currently only by the two simple simple statements). Reloading the navbar would be a bigger operation which I suggest we keep out of this "external" view.